src/player_instances.c

```diff
@@ -254,7 +254,10 @@ long pinstfe_hand_whip(struct PlayerInfo *player, long *n)
       trapst = &game.conf.trapdoor_conf.trap_cfgstats[thing->model];
       if ((trapst->slappable == 1) && trap_is_active(thing))
       {
-          external_activate_trap_shot_at_angle(thing, player->acamera->orient_a, thing_get(player->hand_thing_idx));
+          struct TrapStats* trapstat = &game.conf.trap_stats[thing->model];
+          shotst = get_shot_model_stats(trapstat->created_itm_model);
+          struct Thing* trgtng = ((shotst->model_flags & ShMF_Boulder) != 0) ? INVALID_THING : get_nearest_enemy_creature_in_sight_and_range_of_trap(thing);
```

Comment:
Now you hardcoded it on the shot model, that is not so very nice. Would this not be a lot better to link to TriggerType 1?

Reply:
> As I'm reading it I think it's coded with the Boulder property, not the shot model, isn't?

Yes, it's the Boulder model flag, not the model itself.

Reply:
ok, indeed better. But still begs the question what we want for the other 90-degree triggered traps, of which by default there are none,...

Would be nice I think if at least it looks in the correct general direction you slap in for the 90-degree angles, or do you disagree?

Reply:
I've no idea how that can be done.

Reply:
ok, I will test it like this

Reply:
@AdamPlenty I pushed a commit putting all the power in the hands of the mapmaker. How is that?

Reply:
> @AdamPlenty I pushed a commit putting all the power in the hands of the mapmaker. How is that?

It doesn't work :(. I cannot slap traps whose Slappable is set to 2.

Reply:
it does